### PR TITLE
Create a dedicated gitClient for plan-preview

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/cmd/piped/piped.go [hunk 1]
```diff
@@ -386,8 +386,23 @@ func (p *piped) run(ctx context.Context, t cli.Telemetry) (runErr error) {
...
+		if err != nil {
+			t.Logger.Error("failed to initialize git client for plan-preview", zap.Error(err))
+			return err
+		}
+		defer func() {
+			if err := gc.Clean(); err != nil {
+				t.Logger.Error("had an error while cleaning gitClient for plan-preview", zap.Error(err))
+			} else {
+				t.Logger.Info("successfully cleaned gitClient for plan-preview")
+			}
```
Comment (nakabonne) [inline pkg/app/piped/cmd/piped/piped.go:401]: nit: but it's on you
```suggestion
			if err := gc.Clean(); err != nil {
				t.Logger.Error("had an error while cleaning gitClient for plan-preview", zap.Error(err))
				return
			}
			t.Logger.Info("successfully cleaned gitClient for plan-preview")
```

Comment (nghialv) [reply pkg/app/piped/cmd/piped/piped.go]: Sure. Applied it.

**Comment by nakabonne** [comment]
Cool
/lgtm

**Comment by khanhtc1202** [comment]
🚀 
/approve

**Pull request merged**